### PR TITLE
feat(app): 2393 retry api calls

### PR DESCRIPTION
**Pull request by cy**
- retries errored calls waiting between 1 to 5s
Closes #2393

### CODE REVIEW

File: app/config.js [hunk 1]
```diff
@@ -9,6 +9,6 @@ angular.module("ngApp.config", [])
     "auth_api": "__AUTH__/api",
     "supportedAPI": "1",
     "tag": "https://github.com/NCI-GDC/portal-ui/releases/tag/__VERSION__",
-    "production": __PRODUCTION__,
-    "fake_auth": __FAKE_AUTH__,
+    "production": "__PRODUCTION__",
+    "fake_auth": "__FAKE_AUTH__",
```
Comment (shanewilson) [inline app/config.js]: this still works?

File: app/scripts/core/tests/core.tests.js [hunk 2]
```diff
@@ -1,21 +1,90 @@
...
+  var CoreService,
+      FilesService,
+      ParticipantsService,
+      ProjectsService,
+      AnnotationsService,
+      httpBackend;
 
   // Initialization of the AngularJS application before each test case
-  beforeEach(module('ngApp.files', 'core.services', 'ngProgressLite'));
+  beforeEach(module('ngApp',
```
Comment (shanewilson) [inline app/scripts/core/tests/core.tests.js]: is there any way to break out the interceptor?

File: app/config.js (hunk 1 above)
Comment (cy) [inline app/config.js]: also replacing the quotes https://github.com/NCI-GDC/portal-ui/pull/984/files#diff-b9e12334e9eafd8341a6107dd98510c9R147

File: app/scripts/core/tests/core.tests.js (hunk 2 above)
Comment (cy) [inline app/scripts/core/tests/core.tests.js]: move it to core maybe? I'll give it a shot

File: app/config.js (hunk 1 above)
Comment (shanewilson) [inline app/config.js]: what made this suddenly an issue?

Comment (cy) [inline app/config.js]: tests threw `__PRODUCTION__ undefined` now that it's not loading ngApp though might not be an issue anymore. I'll do some cleanups in the morning.

**Comment by cy** [comment]
moved the error interceptor out of appRun and into core.servies, making the tests much simpler. Checking status code -1 is for testing locally, -1 is returned when the api is turned off. Keep it or remove it? @NCI-GDC/oicrdevs

**Comment by shanewilson** [comment]
could you mock the API response?

**Comment by cy** [comment]
like https://github.com/NCI-GDC/portal-ui/pull/984/files#diff-69d354019ef3c8329f5244f97244eb69R49?

**Comment by cy** [comment]
oh wait, do you mean the status = -1 thing? Yeah I could, in which case I should remove those checks.

**Comment by cy** [comment]
ok hopefully final clean up done. r? @NCI-GDC/oicrdevs

**Comment by alex-wilmer** [comment]
lgtm, though I would prefer template strings over string concat. no deal breaker though.

**Comment by shanewilson** [comment]
👍

**Pull request merged**